flexget/plugins/input/from_telegram.py

```diff
@@ -0,0 +1,219 @@
+                            },
+                            'required': ['fullname'],
+                            'additionalProperties': False,
+                        },
+                        {
+                            'type': 'object',
+                            'properties': {'group': {'type': 'string'}},
+                            'required': ['group'],
+                            'additionalProperties': False,
+                        },
+                    ]
+                },
+            },
+
+            'only_new': {'type': 'boolean','default': True},
+            'entry': {
+                'type': 'object',
+                'properties': {
+                    'url': {'type': 'string', 'format': 'regex'},
+                    'title': {'type': 'string', 'format': 'regex'},
+                },
+                'additionalProperties': {'type': 'string', 'format': 'regex'},
+            },
+        },
+        'required': ['token'],
+        'additonalProperties': False,
+    }
+
+    @plugin.internet(logger)
+    def on_task_input(self, task, config):
+        #Load The Configs
+        token = config['token']
+        only_new = config['only_new']
+        entry_config = config.get('entry')
+        whitelist = config.get('whitelist',[])
+        types = config.get('types')
+
+        #Defaults
+        if not types:
+            types = ['private', 'group']
+
+        #Get Last Checked ID
+        update_id = task.simple_persistence.get(token+'_update_id')
+
+        #Get only new messages
+        params={}
+        if update_id and only_new:
+            update_id += 1
+            params['offset'] = update_id
+        
+        #The Target URL
+        url = "{domain}bot{token}/getUpdates".format(
+            domain=_TELEGRAM_API_URL,
+            token=token
+        )
+
+        #Get Telegram Updates
+        try:
+            response = task.requests.get(
+                url, timeout=60, raise_status=False, params=params
+            ).json()
+        except HTTPError as e:
+            raise plugin.PluginError("Error getting telegram update: {}" % e)
+            return
+
+        #We have a error
+        if not response['ok']:
+            raise plugin.PluginError("Telegram updater returned error {}: {}" % (response['error_code'], response['description'] ) )
+            return
+
+        #Get All New Messages
+        messages = response['result']
+
+        entries = []
+        for message in messages:
+            #This is the ID
+            update_id = message['update_id']
+
+            #Update the last ID for the Bot
+            logger.debug("Last Update set to {}",update_id)
+            task.simple_persistence[token+'_update_id'] = update_id
+            
+            #We Don't care if it's not a message or no text
+            if 'message' not in message or 'text' not in message['message'] or 'chat' not in message['message'] or 'type' not in message['message']['chat']:
```

Comment:
this line is way too long, please run black

Reply:
I assume that you where talking about making a if block, I made a different ident to the if block

Reply:
Black was run